### PR TITLE
Added recipe for nrtk-jatic package

### PR DESCRIPTION
**Pull request by barry-ravichandran**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bjrichardwebster** [comment]
I am willing to be listed

**Comment by barry-ravichandran** [comment]
@conda-forge/help-python , ready for review!

**Comment by barry-ravichandran** [comment]
@conda-forge/help-python 

- The windows build is expected to fail due to the `maite` dependency as per info from this PR: https://github.com/conda-forge/staged-recipes/pull/26303
- `skip: true  # [py==38]` is due to the `nrtk` dependency (most recent versions) building successfully only for `py>=39`

**Comment by barry-ravichandran** [comment]
@conda-forge-admin, please ping conda-forge/help-python

**Comment by eveenhuis** [comment]
I am willing to be listed

File: recipes/nrtk-jatic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+    - poetry-core
+  run:
+    - python
+    - numpy <1.26,>=1.22  # [py>=39 and py<312]
```
Comment (xhochy) [inline recipes/nrtk-jatic/meta.yaml:25]: What's the reason for this numpy pin? Without it, this could be a noarch recipe.

Comment (barry-ravichandran) [reply recipes/nrtk-jatic/meta.yaml:25]: 1. skip: true  # [py==38] is due to the nrtk dependency (https://github.com/conda-forge/nrtk-feedstock) (most recent versions) building successfully only for py>=39
2.  https://github.com/Kitware/nrtk-jatic/blob/ba40077ef165480f1867b2649c612da2e5682dba/pyproject.toml#L37
Here, the constrained dependencies are captured. I tried without the numpy pin and the conda resolver fails to resolve.

**Comment by xhochy** [comment]
You still need to fix/skip the windows build.

**Comment by xhochy** [comment]
Please don't force-push once a recipe was reviewed. This makes the diff harder for the reviewer

**Comment by barry-ravichandran** [comment]
> Please don't force-push once a recipe was reviewed. This makes the diff harder for the reviewer

Sorry about that. I will refrain from force-pushing from now onwards.

**Review by xhochy — APPROVED** [review]

**Pull request merged**